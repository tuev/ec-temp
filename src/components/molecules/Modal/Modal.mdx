import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './Modal.stories.tsx'
import Modal from './Modal.tsx'

<Story name="basic" />
