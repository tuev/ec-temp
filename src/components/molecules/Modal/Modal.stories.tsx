import React, { useState } from 'react'
import Modal from './Modal'
import Button from '../../atoms/Button'
import Typography from '../../atoms/Typography'

export default {
  component: Modal,
  title: 'AppModal',
  includeStories: ['basic'],
}

function BasicModal() {
  const [openModal, setOpenModal] = useState(false)
  return (
    <>
      <Button
        color="primary"
        variant="contained"
        size="small"
        onClick={() => setOpenModal(true)}
      >
        open modal
      </Button>
      <Modal open={openModal} onClose={() => setOpenModal(false)}>
        <>
          <Typography customvariant="header3">My Title</Typography>
          <Typography customvariant="body2">
            My Description Lorem, ipsum dolor sit amet consectetur adipisicing
            elit. Cum eaque soluta illo deleniti unde, cumque sit mollitia,
            incidunt et recusandae, ea odit pariatur officia aliquam minus nulla
            laborum! Laborum, dignissimos! My Description Lorem, ipsum dolor sit
            amet consectetur adipisicing elit. Cum eaque soluta illo deleniti
            unde, cumque sit mollitia, incidunt et recusandae, ea odit pariatur
            officia aliquam minus nulla laborum! Laborum, dignissimos!
          </Typography>
        </>
      </Modal>
    </>
  )
}

export const basic = () => {
  return <BasicModal />
}
