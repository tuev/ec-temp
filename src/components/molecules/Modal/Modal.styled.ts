import { space } from 'styled-system'
import { Modal } from '@material-ui/core'
import styled from 'styled-components'
import { COLOR } from 'theme/colors'
import { BaseModalProps } from './Modal.types'

export const AppModal: React.FC<BaseModalProps> = styled(Modal)<BaseModalProps>`
  ${space}
  display: flex;
  align-items: center;
  justify-content: center;
`

export const AppModalContent: React.FC = styled.div`
  ${space}
  outline: none !important;
  background-color: ${COLOR.white};
  border: 2px solid ${COLOR.black};
  width: 80%;
  max-height: 100%;
  overflow-y: auto;
`
