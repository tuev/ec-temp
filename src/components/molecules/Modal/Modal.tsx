import React, { FC } from 'react'
import { AppModal, AppModalContent } from './Modal.styled'
import { BaseModalProps } from './Modal.types'

const Modal: FC<BaseModalProps> = (props: BaseModalProps) => {
  const { children, ...modalProps } = props
  return (
    <AppModal {...modalProps}>
      <AppModalContent>{children}</AppModalContent>
    </AppModal>
  )
}

export default Modal
