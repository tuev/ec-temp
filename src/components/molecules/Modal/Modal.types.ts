import { ModalProps } from '@material-ui/core/Modal'
import { SpaceProps } from 'styled-system'
export declare type BaseModalProps = ModalProps & SpaceProps
