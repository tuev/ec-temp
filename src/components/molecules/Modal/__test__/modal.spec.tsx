import React, { useState, useCallback } from 'react'
import { mount } from 'enzyme'
import Modal from '..'

// refer test modal
// link: https://medium.com/front-end-weekly/tested-react-build-and-test-modal-using-react-features-and-dom-events-39b7246a3a6f

const WithModal: () => JSX.Element = () => {
  const [openModal, setOpenModal] = useState(false)
  const onClose = useCallback(() => setOpenModal(false), [])
  return (
    <Modal onClose={onClose} open={openModal}>
      <div>Hello World</div>
    </Modal>
  )
}

describe('Modal', () => {
  it('renders modal when open flag is true', () => {
    const props = { open: true }
    const wrapper = mount(
      <div id="root">
        <Modal {...props}>
          <div className="title">Modal Title</div>
        </Modal>
      </div>
    )

    expect(wrapper).toMatchSnapshot()

    const isOpen = wrapper.find(Modal).props().open
    expect(isOpen).toEqual(true)

    const title = wrapper.find('.title').text()
    expect(title).toEqual('Modal Title')
  })

  it('closes the Modal when ESC key is pressed', () => {
    const wrapper = mount(<WithModal />)
    const evt = new KeyboardEvent('keydown', ({
      keyCode: 27,
    } as unknown) as KeyboardEventInit)
    document.dispatchEvent(evt)
    expect(wrapper.find(Modal).props().open).toEqual(false)
  })
})
