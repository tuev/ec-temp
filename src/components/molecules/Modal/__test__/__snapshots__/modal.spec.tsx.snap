// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders modal when open flag is true 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  outline: none !important;
  background-color: #fff;
  border: 2px solid #2E2E2E;
  width: 80%;
  max-height: 100%;
  overflow-y: auto;
}

<div
  id="root"
>
  <Modal
    open={true}
  >
    <Styled(Component)
      open={true}
    >
      <ForwardRef(Modal)
        className="c0"
        open={true}
      >
        <ForwardRef(Portal)
          disablePortal={false}
        >
          <Portal
            containerInfo={
              .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  outline: none !important;
  background-color: #fff;
  border: 2px solid #2E2E2E;
  width: 80%;
  max-height: 100%;
  overflow-y: auto;
}

<body
                style="padding-right: 0px; overflow: hidden;"
              >
                <div
                  class="c0"
                  role="presentation"
                  style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
                >
                  <div
                    aria-hidden="true"
                    style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
                  />
                  <div
                    data-test="sentinelStart"
                    tabindex="0"
                  />
                  <div
                    class="c1"
                    tabindex="-1"
                  >
                    <div
                      class="title"
                    >
                      Modal Title
                    </div>
                  </div>
                  <div
                    data-test="sentinelEnd"
                    tabindex="0"
                  />
                </div>
              </body>
            }
          >
            <div
              className="c0"
              onKeyDown={[Function]}
              role="presentation"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "fixed",
                  "right": 0,
                  "top": 0,
                  "zIndex": 1300,
                }
              }
            >
              <ForwardRef(SimpleBackdrop)
                onClick={[Function]}
                open={true}
              >
                <div
                  aria-hidden={true}
                  onClick={[Function]}
                  style={
                    Object {
                      "WebkitTapHighlightColor": "transparent",
                      "backgroundColor": "rgba(0, 0, 0, 0.5)",
                      "bottom": 0,
                      "left": 0,
                      "position": "fixed",
                      "right": 0,
                      "top": 0,
                      "zIndex": -1,
                    }
                  }
                />
              </ForwardRef(SimpleBackdrop)>
              <TrapFocus
                disableAutoFocus={false}
                disableEnforceFocus={false}
                disableRestoreFocus={false}
                getDoc={[Function]}
                isEnabled={[Function]}
                open={true}
              >
                <div
                  data-test="sentinelStart"
                  tabIndex={0}
                />
                <styled.div
                  tabIndex="-1"
                >
                  <div
                    className="c1"
                    tabIndex="-1"
                  >
                    <div
                      className="title"
                    >
                      Modal Title
                    </div>
                  </div>
                </styled.div>
                <div
                  data-test="sentinelEnd"
                  tabIndex={0}
                />
              </TrapFocus>
            </div>
          </Portal>
        </ForwardRef(Portal)>
      </ForwardRef(Modal)>
    </Styled(Component)>
  </Modal>
</div>
`;
